### DISCUSSION TITLE
Thank you!!

### DISCUSSION THREAD

**Discussion by Copperplate** [tsgrgo/windows-update-disabler#3] [General]
Hi, I just wanted to say a big **THANK YOU** for creating this excellent script! Since it disables Tasks I couldn't previously disable, I'm hoping it will permanently stop Windows 10 from updating itself. Before trying your script, I had manually made changes in Services and the Registry, but those changes were automatically reverted after only 6 days. 

When running the script, I did notice one error which said uhssvc is not a valid service.  **Perhaps that service has been removed or renamed in Windows 10 22H2 home edition?** Anyway, I think everything else worked as intended, as all the other Windows Update processes were disabled and the WaaSMedicSvc.dll was successfully renamed in C:\Windows\System32. I noticed a few other WaaS-related files in there, "WaaSAssessment.dll", "WaaSMedicAgent.exe", "WaaSMedicCapsule.dll", and "WaaSMedicPS.dll". Hopefully those files will not cause things to re-enable down the road... :-)

Anyway, thank you again for providing this wonderful gift to Windows users everywhere. Your amazing work is truly appreciated!!!

-Adam

**Comment by tsgrgo** [comment]
Thank you! I didn't expect such a kind message! I'm happy that I could help! I faced the same issue you had – the Windows Update service would always become enabled again after some period. It took me a long time to realize that the Windows Update Medic was responsible for this. The WaaSMedicSvc.dll is the service that periodically checks if wuauserv is running correctly. You only need to rename this file to completely disable the service. The other files are perfectly fine to stay there.

The uhssvc service is part of Microsoft Update Health Tools, which was introduced in kb4023057. It is located at `C:\Program Files\Microsoft Update Health Tools`. If I'm correct, it can be uninstalled just like any other program in the control panel, so you might have already uninstalled it.

Currently, the only things capable of starting wuauserv are the Microsoft Store and some of the Xbox apps. However, they only set it to manual start (after a reboot it still stays stopped), so the service can start if you open these apps (it would still not install updates though). Originally, I wanted to also rename `wuaueng.dll` (the Windows Update service); however, that would break the Microsoft Store.

**Reply by Copperplate** [reply]
Ah, okay! Thanks for clarifying, that's really good info to know. Yes, after running the script, I noticed the "Scheduled Start" task for WindowsUpdate would periodically change from "Disabled" to "Ready", seemingly due to a "One Time" trigger that was enabled (though all other triggers remained disabled). Perhaps that was the Microsoft Store apps or the Xbox apps causing that then. I've disabled GameBar, but we know Microsoft is persistent...

But it's **so good** to see that "Something went wrong" message in Windows Update, LOL. We knew we had to do something drastic after the last "update" Microsoft sneaked in **destroyed** our system and prevented booting into Windows. We had to clean install Windows 10 twice in two days, and that was crazy. I'm keeping an eye now on the ReportingEvents.log file located in C:\Windows\SoftwareDistribution. Even when updates were "paused" on my system, Microsoft was still sneaking in updates for Feedback Hub, VC Libs, Alarms, Calculator, ZuneVideo, Camera, and Photos. But after running your script, the log is now super short and no more installs are showing... yay! :-)

Have a great week ahead!


P.S. My Dad and I release original music videos from time to time. Feel free to check out our latest ones on [our YouTube channel](https://www.youtube.com/@HeavenInspiredMovies) if you're interested. :-)

**Reply by tsgrgo** [reply]
Thanks for the update! You are right; I haven't noticed it yet, but the "Scheduled Start" also got enabled for me. I'll have to look into this again to figure out what is doing this, I'm pretty sure this behavior came in one of the latest updates. Other than using some resources when the task fires, I think it shouldn't cause any problems.

I'll definitely check out your YouTube channel. It's great that you and your dad share your original creations. 
Wishing you a wonderful week ahead as well!
